player/lua/console.lua

```diff
@@ -1304,6 +1304,72 @@ local function handle_choice_completion(option, before_cur, path_pos)
     return info.choices, before_cur
 end
 
+local function command_flags_at_1st_argument_list(command)
+    local flags = {
+        ['playlist-next'] = {'weak', 'force'},
+        ['playlist-play-index'] = {'current'},
+        ['playlist-remove'] = {'current'},
+        ['rescan-external-files'] = {'reselect', 'keep-selection'},
+        ['revert-seek'] = {'mark', 'mark-permanent'},
+        ['screenshot'] = {'subtitles', 'video', 'window', 'each-frame'},
+        ['stop'] = {'keep-playlist'},
+    }
+    flags['playlist-prev'] = flags['playlist-next']
+    flags['screenshot-raw'] = flags.screenshot
+
+    return flags[command]
+end
+
+local function command_flags_at_2nd_argument_list(command)
+    local flags = {
+        ['apply-profile'] = {'default', 'restore'},
+        ['loadfile'] = {'replace', 'append', 'append-play', 'insert-next',
+                        'insert-next-play', 'insert-at', 'insert-at-play'},
+        ['screenshot-to-file'] = {'subtitles', 'video', 'window', 'each-frame'},
+        ['seek'] = {'relative', 'absolute', 'absolute-percent',
+                    'relative-percent', 'keyframes', 'exact'},
+        ['sub-add'] = {'select', 'auto', 'cached'},
+        ['sub-seek'] = {'primary', 'secondary'},
+    }
+    flags.loadlist = flags.loadfile
+    flags['audio-add'] = flags['sub-add']
+    flags['video-add'] = flags['sub-add']
+    flags['sub-step'] = flags['sub-seek']
+
+    return flags[command]
+end
+
+local function list_executables()
+    local executable_map = {}
+    local path = os.getenv('PATH') or ''
+    local separator = platform == 'windows' and ';' or ':'
+
+    for directory in path:gmatch('[^' .. separator .. ']+') do
+        for _, executable in pairs(utils.readdir(directory, 'files') or {}) do
```

Comment:
Is this really useful? I would rather avoid doing external completions.

Reply:
What do you mean with external? It already completes file paths for commands like loadfile.

Reply:
> What do you mean with external?

As a not relevant to normal mpv operation.

Reply:
For what? What are you doing in your mpv console?

Reply:
e.g. `run wl-copy ${=time-pos}` (copy timestamp to pass it to ffmpeg),  `run mv ${path} some_dir`, `run vlc ${path}` (compare the video in another player), `run rm ${path}`, `run reset` (I need this sometimes when terminal output gets stuck)

Reply:
`run wl-copy ${=time-pos}` shouldn't this be implemented by native property now?
`run vlc ${path}` 😂
`run mv ${path} some_dir`, `run rm ${path}`. You need to autocomplete mv/rm?

Anyway, it's fine if you need it, but I don't see a need personally.

Reply:
wayland clipbard doesn't support writing yet. But yeah wl-copy is probably the main use. You're not gonna use it every day but it's nice in the rare occasion that you need it, and it is just one function.

Reply:
I also run `ff{mpeg,probe} ${file}` sometimes...

Reply:
Lazy to retype the path when it's open in mpv already. Usually it's some sample I download from a ticket into `/tmp` and it has a gibberish name like `a2rdij356gbi22gb.mp4` and there are multiple files with similar human unreadable names.

It's not a big deal to not have it but it's still nice to have it for completeness sake.